File: jja6312/2월21일/피보나치 수.java
class Solution {
    public int solution(int n) {
        // 0 = 0
        // 1 = 0+1
        // 2 = 0+1
        // 3 = 1+1

        int[] arr = new int[n + 1];
        arr[0] = 0;
        arr[1] = 1;
        for (int i = 2; i <= n; i++) {
            arr[i] = arr[i - 2] % 1234567 + arr[i - 1] % 1234567;
        }

        int answer = arr[n] % 1234567;
        return answer;
    }
}

File: zuzubibi/2월/2월 21일/피보나치 수.py
def solution(n):
    arr = [0,1]
    for i in range(2, n+1):
        arr.append((arr[i-1] + arr[i-2]) %1234567)
    return arr[-1]

def anothersolution(n):
    a,b = 0,1
    for _ in range(n):
        a,b = b, (a+b)%1234567
    return a
print(anothersolution(3)) # 2
print(anothersolution(5)) # 5